lib/Launcher.js

```diff
@@ -270,18 +270,18 @@ class Launcher {
   }
 
   /**
-   * @param {!(BrowserOptions & {browserWSEndpoint: string})} options
+   * @param {!(BrowserOptions & {browserWSEndpoint: string, transport: ?Puppeteer.ConnectionTransport})} options
    * @return {!Promise<!Browser>}
    */
   async connect(options) {
     const {
       browserWSEndpoint,
       ignoreHTTPSErrors = false,
       defaultViewport = {width: 800, height: 600},
+      transport = null,
       slowMo = 0,
     } = options;
-    const transport = await WebSocketTransport.create(browserWSEndpoint);
-    const connection = new Connection(browserWSEndpoint, transport, slowMo);
+    const connection = new Connection(browserWSEndpoint, transport || (await WebSocketTransport.create(browserWSEndpoint)), slowMo);
```

Comment:
TODO(einbinder): node6 transpiler broke here.

Reply:
fixed this one.